### ISSUE TITLE
non-trainable Model within custom loss doesn't behave as when it was trained

### ISSUE THREAD

**Issue by LuCeHe** [keras-team/keras#10553]
I've created a custom loss funtion that has to load a fixed model within (F-model). The F-model is a discriminator that assigns a value to the output of the learning model (L-model). The ouput of the F-model should be either really close to -1 or really close to +1 since it's doing binary classification of y_true. The fact is that it is giving back numbers continuously distributed between -.3 and -.8, and the range changes every time I run my experiments.

``` python

def discriminator_loss(y_true, y_pred):
    
    weightsFilename = 'weights-file.h5'
    modelFilename = os.path.join(CDIR, weightsFilename)
    
    discriminator_model = buildDiscriminator()
    discriminator_model.load_weights(modelFilename, by_name=True)
    discriminator_model.trainable = False
    
    # y_true is one_hot, back to indexes with argmax
    x_enc = K.argmax(y_true, axis=-1)  

    # the discriminator is a seq2scalar
    ynew = discriminator_model(x_enc)
    dis_loss = K.mean(K.mean(ynew))

    return dis_loss

```

To try to understand the issue, I wrote the loss in numpy, and funny enough, when I load the model iteratively in a loop, it reproduces the error that keras was giving, while when I charge the network once before the loop it works perfectly:

    
    
``` python    

def discriminator_loss_np():
    
    i = 0
    for y in generator(64):
        weightsFilename = 'weights-file.h5'
        modelFilename = os.path.join(CDIR, weightsFilename)
        
        discriminator_model = buildDiscriminator()
        discriminator_model.load_weights(modelFilename, by_name=True)
        discriminator_model.trainable = False
        
        i += 1
        
        x = np.argmax(y, axis=-1)  

        ynew = discriminator_model.predict(x)
        dis_loss = np.mean(np.mean(ynew))
        print(dis_loss)
        
        if i == 100: break
```

My hypothesis is that keras is not preloading the F-model at the beginning and is loading it everytime it's calling the loss. What's the way around??
